airflow/ui/src/components/FlexibleForm/SelectElement.tsx

```diff
@@ -0,0 +1,120 @@
+ * under the License.
+ */
+import type { ParamSchema, ParamSpec } from "src/queries/useDagParams";
+
+import type { FlexibleFormElementProps } from ".";
+import { FieldAdvancedArray } from "./FieldAdvancedArray";
+import { FieldBool } from "./FieldBool";
+import { FieldDate } from "./FieldDate";
+import { FieldDateTime } from "./FieldDateTime";
+import { FieldDropdown } from "./FieldDropdown";
+import { FieldMultiSelect } from "./FieldMultiSelect";
+import { FieldMultilineText } from "./FieldMultilineText";
+import { FieldNumber } from "./FieldNumber";
+import { FieldObject } from "./FieldObject";
+import { FieldString } from "./FieldString";
+import { FieldStringArray } from "./FieldStringArray";
+import { FieldTime } from "./FieldTime";
+
+const inferType = (param: ParamSpec) => {
+  if (Boolean(param.schema.type)) {
+    // If there are multiple types, we assume that the first one is the correct one that is not "null".
+    // "null" is only used to signal the value is optional.
+    if (Array.isArray(param.schema.type)) {
+      return param.schema.type.find((type) => type !== "null") ?? "string";
+    }
+
+    return param.schema.type ?? "string";
+  }
+
+  // If the type is not defined, we infer it from the value.
+  if (Array.isArray(param.value)) {
+    return "array";
+  }
+
+  return typeof param.value;
+};
+
+const isFieldAdvancedArray = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "array" && fieldSchema.items?.type !== "string";
+
+const isFieldBool = (fieldType: string) => fieldType === "boolean";
+
+const isFieldDate = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "string" && fieldSchema.format === "date";
+
+const isFieldDateTime = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "string" && fieldSchema.format === "date-time";
+
+const enumTypes = ["string", "number", "integer"];
+
+const isFieldDropdown = (fieldType: string, fieldSchema: ParamSchema) =>
+  enumTypes.includes(fieldType) && Array.isArray(fieldSchema.enum);
+
+const isFieldMultilineText = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "string" && fieldSchema.format === "multiline";
+
+const isFieldMultiSelect = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "array" && Array.isArray(fieldSchema.examples);
+
+const isFieldNumber = (fieldType: string) => {
+  const numberTypes = ["integer", "number"];
+
+  return numberTypes.includes(fieldType);
+};
+
+const isFieldObject = (fieldType: string) => fieldType === "object";
+
+const isFieldStringArray = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "array" &&
+  (!fieldSchema.items || fieldSchema.items.type === undefined || fieldSchema.items.type === "string");
+
+const isFieldTime = (fieldType: string, fieldSchema: ParamSchema) =>
+  fieldType === "string" && fieldSchema.format === "date";
+
+export const SelectElement = ({ key, name, param }: FlexibleFormElementProps) => {
```

Comment:
Let's rename this. I thought it was a dropdown select at first.

Maybe `FieldInput`?

Reply:
Named it ` FieldSelector` - is that OK and describes what it does?